### ISSUE TITLE
Some problems occurred during the training

### ISSUE THREAD

**Issue by winnieepooh** [jantic/DeOldify#479]
Hi,I have some questions during the training, and I'm very grateful for your answers.
Firstly,learn_gen.freeze() only freezes the conv layer, not the BN layer.Do I need to freeze the BN layer separately?
Secondly,after training the 192px portion, the resulting image was then put into the GAN for training, but why was it clean when it was put in and ended up with a lot of dirty data?
Training too many times always do not get better.  
I look forward to your advice.
Thanks!
input:![image](https://github.com/jantic/DeOldify/assets/94221441/562138ba-ee94-43fd-a277-5bc7c746f9ed)
output:
![ILSVRC2012_test_00071662](https://github.com/jantic/DeOldify/assets/94221441/42320a6f-1b27-41a2-8e2a-8707e4072dde)

**Comment by weirui0430** [comment]
请问怎么训练呀

**Comment by jantic** [comment]
> Firstly,learn_gen.freeze() only freezes the conv layer, not the BN layer.Do I need to freeze the BN layer separately?

That was a deliberate design decision by FastAI. I honestly forget what the exact justification was for it, but it was deliberate. You can modify the source code there of the FastAI fork there if you want to experiment with it of course!

> Secondly,after training the 192px portion, the resulting image was then put into the GAN for training, but why was it clean when it was put in and ended up with a lot of dirty data?

So what I'm seeing there is evidence of instability of training.  Especially the blue blobs and the b/w images. Top candidates for causing this are: 
1. Changing the batch size without modifying the learning rate accordingly, or vice versa. Generally you want to halve the learning rate if you halve the batch size, and the other way around if you double it.
2. Using fp16 (mixed precision) training naively. I've never tried mixed precision with this particular repo, but I did with my commercial stuff and I can tell you there's an art to doing it.
3. Changing anything with the architecture without experimenting with updates to learning rate via learning rate finder.
4. Changing the optimizer.

Hope that helps!

1.
